### ISSUE TITLE
Error in forward/backward

### ISSUE THREAD

**Issue by Withhite14** [alexandrosstergiou/progressive-action-prediction#7]
I have an error when start training 
INFO one of the variables needed for gradient computation has been modified by an inplace operation: [torch.cuda.FloatTensor [4096, 256]], which is output 0 of ViewBackward, is at version 1; expected version 0 instead. Hint: enable anomaly detection to find the operation that failed to compute its gradient, with torch.autograd.set_detect_anomaly(True).
WARNING Error in forward/backward: forward executed: True , backward executed: False
WARNING Creating dataloader for batch of size (2,16,224,224)

after research i found the solution about clone the data before network but i can't find where to clone

Thank you!

**Comment by JinniPi** [comment]
@Withhite14 I got the same issue. You can check the version of torch.  I set torch==1.11.0+cu113, python=3.9. In loss update I set backward(retain_graph=True). then the issue is fixed

**Comment by Withhite14** [comment]
> @Withhite14 I got the same issue. You can check the version of torch. I set torch==1.11.0+cu113, python=3.9. In loss update I set backward(retain_graph=True). then the issue is fixed

Thanks for your answer. I change the version of torch to torch==1.11.0+cu113 and python3.9 but it doesn't work for me

**Comment by JinniPi** [comment]
> > @Withhite14 I got the same issue. You can check the version of torch. I set torch==1.11.0+cu113, python=3.9. In loss update I set backward(retain_graph=True). then the issue is fixed
> 
> Thanks for your answer. I change the version of torch to torch==1.11.0+cu113 and python3.9 but it doesn't work for me
What is version cuda in your computer?

**Comment by Withhite14** [comment]
> What is version cuda in your computer?

cuda 11.3 with cudnn8.7 on ubuntu20.04 and I use MTNet_xs as a backbone

**Comment by JinniPi** [comment]
@Withhite14  
I set the version of all requirements as in readme. After that, I install pytorch torch==1.11.0+cu113, python=3.9. When I run that code "_python train.py --config config/UCF-101/latents_256/conf.yml --video_per 0.3 --num_samplers 3 --gpus 0_"  the other loss showed, and I come to the line 615-622 in model.py set  I set backward(retain_graph=True). It is fixed. Maybe you should try to update the new version of torch and ensure that all requirements are installed and activate your environment. For example Ada pool

**Comment by Withhite14** [comment]
> @Withhite14 I set the version of all requirements as in readme. After that, I install pytorch torch==1.11.0+cu113, python=3.9. When I run that code "_python train.py --config config/UCF-101/latents_256/conf.yml --video_per 0.3 --num_samplers 3 --gpus 0_" the other loss showed, and I come to the line 615-622 in model.py set I set backward(retain_graph=True). It is fixed. Maybe you should try to update the new version of torch and ensure that all requirements are installed and activate your environment. For example Ada pool

After I reinstall everything and change environment the same as u suggest I finally got it
Thank you so much for helping
